dataprovider/data_provider.go

```diff
@@ -20,11 +20,13 @@ package dataprovider
 import (
 	"github.com/elastic/beats/v7/libbeat/beat"
 
-	"github.com/elastic/cloudbeat/dataprovider/types"
 	"github.com/elastic/cloudbeat/resources/fetching"
 )
 
 type CommonDataProvider interface {
-	FetchData(resource string, id string) (types.Data, error)
```

Comment:
why not add `GetId` under the `CommonDataProvider` interface?

Reply:
tl;dr: Single responsibility.
This interface is about to remove completely and we are going to have the new version:
```
type ElasticCommonDataProvider interface {
	GetElasticCommonData() (map[string]interface{}, error)
}
```
The new interface will have different implementations which will add fields in different layer of the resource. (resource specific, cloud specific, benchmark specific, global).
So while I'm in the process of removing that interface I don't want to add more functionality to it.